torch/lib/THCUNN/generic/FusedRNNKernel.cu

```diff
@@ -41,16 +41,21 @@ int THNN_(minIndexType)(THCState *state, int count, ...)
   va_list list;
   va_start(list, count);
 
-  int maxDim = -2;
-  for (int arg=0; arg < count; ++arg){
-    THCTensor* tens = va_arg(list, THCTensor*);
-    if(THCTensor_(isContiguous)(state, tens)) continue;
-    int tensdims = TensorUtils<THCTensor>::getDims(state, tens);
-    maxDim = (( tensdims> maxDim) ? tensdims : maxDim);
+  THCTensor* tens = va_arg(list, THCTensor*);
+  int startDim = TensorUtils<THCTensor>::getDims(state, tens);
+  bool canCollapse = THCTensor_(isContiguous(state,tens));
+
+  for (int arg=1; arg < count; ++arg){
+    tens = va_arg(list, THCTensor*);
+    canCollapse = canCollapse && THCTensor_(isContiguous)(state, tens);
+    if(TensorUtils<THCTensor>::getDims(state, tens) != startDim){
+      va_end(list);
+      return -1;
+    }
   }
-
   va_end(list);
-  return maxDim;
+  if(canCollapse) return -2;
+  return startDim;
```

Comment:
Since I'm not templating  the indexing on every tensor that I put in I need to agree on a common DIM to template on. If all tensors can collapse to 1D do so, if not they all need to be either 1 or 2. Otherwise need to run as -1 DIM.